common/logf.c

```diff
@@ -268,6 +273,105 @@ logf_handler_set_prio(logf_handler_t *handler, logf_prio_t prio)
 	handler->prio = prio;
 }
 
+static int
+logf_current_log_cb(const char *path, const char *file, void *data)
+{
+	list_t **log_file_list = (list_t **)data;
+
+	str_t *path_str = str_new(path);
+	str_append(path_str, "/");
+	str_append(path_str, file);
+
+	//only process .current symlinks.
+	if ((strstr(file, ".current") == NULL) || (!file_is_link(str_buffer(path_str)))) {
+		return 0;
+	}
+
+	char buffer[PATH_MAX];
+
+	//-1 is used to leave space for null terminator.
+	ssize_t len = readlink(str_buffer(path_str), buffer, PATH_MAX - 1);
+
+	if (len != -1) {
+		// Ensure the buffer is null-terminated
+		buffer[len] = '\0';
+		DEBUG("Adding currently used logfile to list %s", buffer);
+		*log_file_list = list_append(*log_file_list, mem_strdup(buffer));
+	} else {
+		ERROR_ERRNO("readlink");
+		return -1;
+	}
+
+	str_free(path_str, true);
+
+	return 0;
+}
+
+list_t *
+logf_get_currently_used_log_files_new(const char *path)
+{
+	list_t *current_log_files = NULL;
+	int dir_ret = dir_foreach(path, &logf_current_log_cb, &current_log_files);
+
+	if (dir_ret < 0) {
+		WARN("Something went wrong during traversal of LOGFILE_DIR");
```

Comment:
plz change this to ERROR with an message including the path, e.g.
ERROR("An error occured during traversal of %s", path"

Reply:
actual path ist still not included in log message: "... %s", path